### PR TITLE
Avoid MantidMatrix changing the number display format

### LINKED ISSUE #20956: MantidMatrix display format changes on update
### Expected behavior
When a workspace is updated, the number format/precision displayed by MantidMatrix should remain the same/default/sensible.

### Actual behavior
If a workspace is updated while a MantidMatrix is already open, the number format changes to scientific notation, whereas the default (decimal) format is usually more sensible.

### Steps to reproduce the behavior
- Create a workspace using this script:
```CreateWorkspace(OutputWorkspace='ws', DataX='1,2', DataY='10.123', DataE='0.5', NSpec=1, UnitX='TOF')```
- In MantidPlot, double-click the workspace to show the MantidMatrix. The Y value is displayed as `10.123`
- Re-run the script to re-create the workspace. The Y value display is changed to `1.012300e+01`.

### Platforms affected
All

### PR DESCRIPTION
**Pull request by gemmaguest**
This PR adds a fix to keep the number format used by `MantidMatrix` the same when the underlying workspace is updated - previously, the format was being changed to scientific notation when the workspace is updated.

**To test:**
Follow the instructions in the linked issue. The number format should no longer change to scientific notation.

Fixes #20956 

**Release Notes** 
Added to the release notes.

---

#### Reviewer ####

Please comment on the following ([full description](http://www.mantidproject.org/Individual_Ticket_Testing)):

##### Code Review #####

- [ ] Is the code of an acceptable quality?
- [ ] Does the code conform to the [coding standards](http://www.mantidproject.org/Coding_Standards)?
- [ ] Are the unit tests small and test the class in isolation?
- [ ] If there are changes in the release notes then do they describe the changes appropriately?

##### Functional Tests #####

- [ ] Do changes function as described? Add comments below that describe the tests performed?
- [ ] Do the changes handle unexpected situations, e.g. bad input?
- [ ] Has the relevant (user and developer) documentation been added/updated?

Does everything look good? Mark the review as **Approve**. A member of `@mantidproject/gatekeepers` will take care of it.

### CODE REVIEW

File: MantidPlot/src/Mantid/MantidMatrix.cpp [hunk 1]
```diff
@@ -1326,17 +1332,24 @@ void MantidMatrix::setupNewExtension(MantidMatrixModel::Type type) {
...
         auto &extension = it->second;
         extension.model = new MantidMatrixModel(this, ws.get(), m_rows, m_cols,
-                                                m_startRow, it->first);
+                                                m_startRow, type);
         connectTableView(extension.tableView.get(), extension.model);
+        auto format = m_extensionRequest.getFormat(
+            type, m_extensions, MantidPreferences::MantidMatrixNumberFormatY());
+        auto precision = m_extensionRequest.getPrecision(
+            type, m_extensions,
+            MantidPreferences::MantidMatrixNumberPrecisionY());
```
Comment (edwardb96) [inline MantidPlot/src/Mantid/MantidMatrix.cpp]: Should these be `MantidMatrixNumberPrecisionDx` and `MantidMatrixNumberFormatDx` instead?

**Review by edwardb96 — CHANGES_REQUESTED** [review]
This seems to work, just one potential C+P error.

File: MantidPlot/src/Mantid/MantidMatrix.cpp (hunk 1 above)
Comment (gemmaguest) [reply MantidPlot/src/Mantid/MantidMatrix.cpp]: Yes, this was copied from the code in setupNewExtension, which should also have been using the `Dx` versions!

**Comment by gemmaguest** [comment]
You can test that the X errors tab gets created/updated ok with this script. As per the original test instructions, run this once, then double-click the workspace in MantidPlot to display it, then run this script again to update it. (Note that there is a separate bug where the X errors tab displays one too many columns, see #20972).
```
CreateWorkspace(OutputWorkspace='ws', DataX='1,2', DataY='10.123', DataE='0.5', NSpec=1, UnitX='TOF')
for i in range(0,len(mtd['ws'].dataDx(0))):
    mtd['ws'].dataDx(0)[i] = 0.5 
```

**Comment by gemmaguest** [comment]
@edwardb96 This is ready for another look

**Review by edwardb96 — APPROVED** [review]

**Pull request merged**